### DISCUSSION TITLE
Feature Request: Show a "Year" view of events

### DISCUSSION THREAD

**Discussion by mcolinsk** [vkurko/calendar#293] [Ideas]
The goal: In a quick glance, show what days have events for an entire year.

it would be nice to show all events in a "year" view. The view would be a vertical scollable viewport that fits 2, 3 or 4 months wide. Days with events would show a coloured dot or bar on the bottom of date square. 
The look can similar to ios Calendar app, or google calendar.

The width of events should be somewhat fixed. On smaller screens, show less months wide. Probably just use display flex, and allow flex-wrapping.

See below for a quick sample. I just edited your example for todays date, updated css styles and copied the month of july into 12 months.

![Screenshot 2024-07-03 at 11 56 01 AM](https://github.com/vkurko/calendar/assets/48663709/67c69ffd-ee79-4ba3-b42c-dc2a44209678)

See the attached zip file. You can inspect the css I used to create the example.
[Event Calendar Sample.html.zip](https://github.com/user-attachments/files/16087605/Event.Calendar.Sample.html.zip)

**Comment by vkurko** [comment]
Thanks for the idea. Any assistance in the implementation of such a view, as well as [sponsorship](https://github.com/sponsors/vkurko), is welcome.

**Reply by mischasigtermans** [reply]
To extend the request of the year view, I'm in need of this view:
![](https://user-images.githubusercontent.com/379534/68153237-2adb7e00-ff13-11e9-8945-3bb6d07fe6ba.png)

@vkurko what sponsorship are we talking about for this?

**Reply by vkurko** [reply]
You can become a [sponsor](https://github.com/sponsors/vkurko), which will support the further development of the calendar. Sponsors' opinions on popular feature requests are also taken into account with a certain priority.

**Comment by pauloffb** [comment]
FullCalendar has it like multimonth allowing to customize the grid and the number of months which I think is even better than a year view:

[Docs](https://fullcalendar.io/docs/multimonth-grid)
[Demo](https://fullcalendar.io/docs/multimonth-grid-demo)

**Reply by mischasigtermans** [reply]
Thanks for sharing. Not what we’re after though, we need to see all weekends aligned in a few to determine availability.

**Reply by lgrkvst** [reply]
@mischasigtermans I'm intrigued. That's a really interesting view. Could you tell me the use-case in which a view like this is preferable?

**Reply by mischasigtermans** [reply]
Managed to get this sorted. This is what I needed:
<img width="2878" height="1094" alt="screenshot-005900 (2026-04-16)" src="https://github.com/user-attachments/assets/9bc896f3-ed90-4afe-8022-68a6e55f9e38" />
